### ISSUE TITLE
Create FAQ / ELI5 section

### ISSUE THREAD

**Issue by abitmore** [bitshares/dev.bitshares.works#1]

**Comment by abitmore** [comment]
Some useful conversation recorded below.

Nobody
libraries/fc/CMakeFiles/fc.dir/build.make:1286: recipe for target 'libraries/fc/CMakeFiles/fc.dir/src/crypto/base58.cpp.o' failed
make[2]: * [libraries/fc/CMakeFiles/fc.dir/src/crypto/base58.cpp.o] Error 1
CMakeFiles/Makefile2:214: recipe for target 'libraries/fc/CMakeFiles/fc.dir/all' failed
make[1]: * [libraries/fc/CMakeFiles/fc.dir/all] Error 2
Makefile:127: recipe for target 'all' failed
make: *** [all] Error 2


I get this error during make
LIke when I installed dependencies on a separate node and tried this git clone https://github.com/bitshares/bitshares-core.git
cd bitshares-core
git checkout <LATEST_RELEASE_TAG>
git submodule update —init —recursive
cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo .
make

GitHub
bitshares/bitshares-core
bitshares-core - BitShares: Your share in the Decentralized Exchange
2.0.180425 - used this release tag

Digital Lucife:registered:
Nobody
libraries/fc/CMakeFiles/fc.dir/build.make:1286: recipe for target 'libraries/fc/CMakeFiles/fc.dir/src/crypto/base58.cpp.o' faile
longer output pls
Nobody
2.0.180425 - used this release tag
use master
git checkout master
Nobody
[Photo]

Digital Lucife:registered:
cmake -DCMAKE_BUILD_TYPE=Release .
Nobody
Digital Lucife:registered:
git checkout master
ok will do that

はるか
it seems that you are using openssl 1.1
you will need 1.0.x at the moment
Nobody
はるか
you will need 1.0.x at the moment
how did you find that that the issue, that obvious from the screenshot?

はるか
BIGNUM is provided by openssl, and openssl 1.1 has some api changes
openssl 1.1 compability is a known issue at the moment
Nobody
ok ok

Alex M - clockwork
Digital Lucife:registered:
git checkout master
quite sure master is always latest release tag  so it's the same

Digital Lucife:registered:
Alex M - clockwork
quite sure master is always latest release tag so it's the same
Since public API use requires master/latest to communicate with blockchain, i found it quite useful using master instead of release ver, especially when we are doing update :)
Nobody
ok ok cleared the issue and now its under make ..
I hope I get this up and running today

Alex M - clockwork
Nobody
ok ok cleared the issue and now its under make .. I hope I get this up and running today
hwo much RAM in the machien building it?
Nobody
64 gb

Alex M - clockwork
ok

Digital Lucife:registered:
Nobody
ok ok cleared the issue and now its under make .. I hope I get this up and running today
:)
Nobody
So after make, the below command will do right?

programs/witness_node/witness_node -s «ip-of-the-running-node:61705» —rpc-endpoint 127.0.0.1:8090
Then after syncing,

programs/cli_wallet/cli_wallet -s ws://«ip-of-the-running node»:8090 -H «ip-of-the-running-node»:8091 ??

Digital Lucife:registered:
Nobody
So after make, the below command will do right? programs/witness_node/witness_node -s «ip-of-the-running-node:61705» —rpc-endpo
Are y doing local or public node ?
Nobody
public node

Digital Lucife:registered:
edit bitshares-core/programs/witness_node/witness_node_data_dir/config.ini
p2p-endpoint = 0.0.0.0:1776
rpc-endpoint = 0.0.0.0:8090
start just ./witness_node
127.0.0.1 - loopback adapter - localhost only
Nobody
there's no witness_node_data_dir currently present 

So I just create it?

Digital Lucife:registered:
no
go to
bitshares-core/programs/witness_node
Nobody
[Photo]

Digital Lucife:registered:
and start ./witness_node
wait 5 secs
CTRL+C
Nobody
alright

Digital Lucife:registered:
then ull have folder and config.ini inside
Nobody
[Photo]
Actually I am getting an error
but yes the directory is created

Digital Lucife:registered:
edit config.ini as instructed
and light it up again
make sure that you're not running any other instance of witness_node process
Nobody
Digital Lucife:registered:
127.0.0.1 - loopback adapter - localhost only
This part I dont get it!
did the rest of the changes
starting witness_node again

Alex M - clockwork
Nobody
This part I dont get it!
if you want to access your node ONLY from the same machine, use 127.0.0.1 as adapter
if you want it to be publicly available
use 0.0.0.0
Nobody
ok ok cool

Alex M - clockwork
depends on your setup really
Nobody
[Photo]
I hope the first line is okay?

Then there was another red line specifying I havent mentioned a witness
[Photo]
But otherwise is running now

Digital Lucife:registered:

Nobody
Photo
Its cool. Its config only for block producers.

Nobody
Photo
And you are syncing now!
Nobody
the ./witness_node is going on

Digital Lucife:registered:
For now, try to be always within folder witness_node when you are starting ./witness_node
we will have soon sh script with working path
to make it a bit more simple for builds from source

Alex M - clockwork
are you running it inside screen?
Nobody
Digital Lucife:registered:
For now, try to be always within folder witness_node when you are starting ./witness_node
Okay cool

Digital Lucife:registered:
Alex M - clockwork
are you running it inside screen?
I do

Alex M - clockwork
Digital Lucife:registered:
I do
i meant nobody :)
Nobody
Alex M - clockwork
are you running it inside screen?
tmux

Alex M - clockwork
ok
keep in mind you wont be able to use teh api until the node is synced
which might take a while
Nobody
Yeah Yeah I understand that
Nobody
Then after syncing, programs/cli_wallet/cli_wallet -s ws://«ip-of-the-running node»:8090 -H «ip-of-the-running-node»:8091 ??
after syncing then how do I create a wallet for the given config? @murda_ra

Digital Lucife:registered:
./cli_wallet -s ws://ip:8090
Nobody
cool
Nobody
Ok lets say I installed the bitshares-ui on the same server 

and suppose I change the 0.0.0.0 to localhost, will I be able to connect to the node using the UI?
And how long will 9TB hold up for running a public full node?
How would I be required to maintain if this were to go on forever?

Alex M - clockwork
Nobody
Ok lets say I installed the bitshares-ui on the same server and suppose I change the 0.0.0.0 to localhost, will I be able to c
bitshares-ui is served from anywhere but runs locally in your browser...so it needs a remote node to connect to
so your node should be publicly available
Nobody
And how long will 9TB hold up for running a public full node? How would I be required to maintain if this were to go on forever?
HDD space thats fine for quite a while
RAM is the issue
depends on amount of accoutn history you hodl etc.
Nobody
Alex M - clockwork
so your node should be publicly available
Cool so running it on localhost is only for testing/developemental purposes?

Alex M - clockwork
Nobody
Cool so running it on localhost is only for testing/developemental purposes?
or anything else you might plan to use it for
Nobody
Alex M - clockwork
depends on amount of accoutn history you hodl etc.
Can you elaborate?

Alex M - clockwork
https://github.com/bitshares/bitshares-core/wiki/Memory-reduction-for-nodes

GitHub
bitshares/bitshares-core
bitshares-core - BitShares: Your share in the Decentralized Exchange
see here
Nobody
Alex M - clockwork
or anything else you might plan to use it for
Like ?
Alex M - clockwork
https://github.com/bitshares/bitshares-core/wiki/Memory-reduction-for-nodes
ok ill see
thankyou

Alex M - clockwork
mayeb you run a bot on your server and want to interface to a local ndoe for low latency...you dont want anyone else to have access to it
so your node only listens at localhost
Nobody
ok ok awesome got that!
Alex M - clockwork
https://github.com/bitshares/bitshares-core/wiki/Memory-reduction-for-nodes
So if i were to change any of those configs now or later on

would simply ctrl+c and re running ./witness_node do it?

Or would I be required to do something else?

Alex M - clockwork
stop and rerun should be ok
Nobody
cool
Nobody
Digital Lucife:registered:
./cli_wallet -s ws://ip:8090
First thing to do is setting up a password for the newly created wallet prior to importing any private keys::

»> set_password <password>
null
»> unlock <password>
null
»>
Wallet creation is now done.



So I need to do this?

my blockchain is not synced it!
Im just asking

Once its synced I open another tab and run that command

meaning Im opening up an ability for other users to create a wallet right?
where does this pwd come into play

Alex M - clockwork
cli_wallet is a command line utility wallet
it creates a local password protected file (that's what the password is for) to hold any keys you import into it
its not necessary for using/running the node
only if you want to interact with your account via command line
Nobody
ok got it !

Kim StaflunD
Alex M - clockwork
only if you want to interact with your account via command line
1% cmd OF bitshares users, rest 99% ui wallet... i would guess....
Nobody
cool
i would recommend you start monitor the node from day one, in and out going network traffic, so you learn what happen under the hood... When error pop....
Nobody
ok ok I will

Digital Lucife:registered:
Alex M - clockwork
it creates a local password protected file (that's what the password is for) to hold any keys you import into it
[👍 Sticker]
Nobody
[Photo]
May I know what this pay with account means?
How do I log out and create another account like I did the first time?

Or why does this functionality exists?

Digital Lucife:registered:

Nobody
Photo
Just open incognito window
Nobody
Okay I understand that, but may I know why it exists in the first place?

Alex M - clockwork
Nobody
Okay I understand that, but may I know why it exists in the first place?
accounts on the dex are not created for free...someone has to pay for them
Nobody
So when I create account for the first time, the network pays for it?

Alex M - clockwork
Nobody
So when I create account for the first time, the network pays for it?
a faucet...set up by whoever runs/hosts the wallet website

Digital Lucife:registered:
Alex M - clockwork
a faucet...set up by whoever runs/hosts the wallet website
Which doesnt mean you can't use existing faucet if you want to run a wallet.
:)

Alex M - clockwork
Digital Lucife:registered:
Which doesnt mean you can't use existing faucet if you want to run a wallet.
ofcourse...faucet operator wants as many users as possible...if they dont actually use his website/nodes to cost him extra infrastructure even better :P
Nobody
[Photo]

Digital Lucife:registered:

Nobody
Photo
There we are

Alex M - clockwork
synced
Nobody
now this means by blockchain is synced right?
which Ill be able to connect to it

using ws://<server-ip>

Alex M - clockwork
yes
Nobody
under personal node
it still says unreachable

Digital Lucife:registered:
yeah, but referal wallet only accept wss: (SSL)

Alex M - clockwork
using wallet.bitshares.org?

Digital Lucife:registered:
only cli_wallet works without SSL
or local hosted UI

Alex M - clockwork
yeah...its reachable under ssl so you need wss for the nodes too
what DL said

Digital Lucife:registered:
You can use letsencrypt for free SSL or you can purchase Premium SSL (If you plan to host your node/domain for a year)
and you will need domain name not just IP
Nobody
Digital Lucife:registered:
and you will need domain name not just IP
Yeah I get that
ok

Digital Lucife:registered:
Nobody
ok
YOU DONT NEED TO STOP WITNESS NODE
to apply SSL
Sorry for caps, just realized nobody mentioned that.

Alex M - clockwork
not if you're proxying to it
in theory you can get the node itself to load certs/keys and work under ssl
but never got it to work :P

Digital Lucife:registered:
Alex M - clockwork
in theory you can get the node itself to load certs/keys and work under ssl
Lets not involve newcomers to our theories for now :) But i would also love to see that SSL handling on core level working
Nobody
So say I have another small site running on the same node on an apache server
and this core is running on it as well

site is running on 8080

for now I simply created a subdomain on it.. to link to the blockchain but I am confused here.
Now for letsencrypt a conf file has to exist right?
usually for a site I point this conf to the public directory, now where do i point this one to?

and I guess I give a subdomain a proxy to <server-ip>:8090 ????

Digital Lucife:registered:
Nobody
So say I have another small site running on the same node on an apache server and this core is running on it as well site is ru
add another site configuration for nginx under /etc/nginx/sites-available/
you can use existing default to make a copy and set another ssl, subdomain, websocket, etc.
node is on 8090
Nobody
Document root?

Digital Lucife:registered:
irellevant
point wherever you want
Nobody
cool

Digital Lucife:registered:
you can create welcome.html
and point it there
when ure done and have it running with ssl
let me know to assist you on wrapping 8090 as /ws /wss in nginx
or bug @clockworkgr if im not here. We are 5 hours different time zone, so you'll probably hit one of us
Nobody
11 eof_exception: End Of File
stringstream
    {}
    th_a  sstream.cpp:66 readsome

    {"str":""}
    th_a  json.cpp:463 from_string


I simply created a index.html inside the main folder and pointed my apache2(instead of nginx) server to it.
When I accessed the subdomain I get this!


yes it has ssl now
and I tried connecting to it via web,-ui still says unreachable!

Digital Lucife:registered:
even says unreachable
now click connect
:D
wss://domain:8090

Abit Moreadmin
Alex M - clockwork
cli_wallet is a command line utility wallet
are these got documented somewhere? ELI5 FAQ or something

Digital Lucife:registered:
Abit More
are these got documented somewhere? ELI5 FAQ or something
not so well and not that organized

Alex M - clockwork
Abit More
are these got documented somewhere? ELI5 FAQ or something
there is a parallel universe timeline where they are , yes.

Digital Lucife:registered:
aahahahahha

Alex M - clockwork
actually
should add an actual ELI5 section in the dev portal :P
literally called ELI5 :P

Abit Moreadmin
IMHO @clockworkgr is the boss of new document site
so, bug him

Alex M - clockwork
ΟΙ!
i'm really not :P

Abit Moreadmin
you can be if you want

Alex M - clockwork
possibly
just not yet :P
im just trying to get thigns moving
if I was, I'd just tell you to open a ticket at the dev.bitshares.works repo :P
(please don't...I know you enjoy it)

Abit Moreadmin
:D
why not

Alex M - clockwork
cause im not getting paid for this shit yet :P
just kidding
go ahead if you think of something
better than trying to remember shit we missed later on
or just edit the plan/structure in the  README...i dont mind

**Comment by clockworkgr** [comment]
You could probably post everything from Dev/Node admins as "useful" conversation when it comes to ELI5

**Comment by clockworkgr** [comment]
Fixed in https://github.com/bitshares/dev.bitshares.works/commit/10eb346b57514ab238269206164bbed57faf1193 :P
